### PR TITLE
fix: #6343 use d.GetOk to get qname_runtime_endpoint

### LINKED ISSUE #6343: Failed to create Cognitive Service Account when qname_runtime_endpoint is not set
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

v2.3.0 or newer

### Affected Resource(s)

* `azurerm_cognitive_account`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
resource "azurerm_cognitive_account" "cognitive_service" {
  name                = var.cognitive_service_name
  resource_group_name = var.resource_group_name
  location            = var.location
  kind                = "CognitiveServices"
  sku_name            = "S0"
}
```

### Expected Behavior

Successfully create cognitive account.

### Actual Behavior

```
azurerm_cognitive_account.cognitive_service: Creating...

Error: Error creating Cognitive Services Account ***: cognitiveservices.AccountsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="ApiPropertiesInvalid" Message="The given 'apiProperties' '{\r\n  \"qnaRuntimeEndpoint\": \"\"\r\n}' is invalid. Validation errors: Property 'qnaRuntimeEndpoint' has not been defined and the schema does not allow additional properties. Path 'apiProperties.qnaRuntimeEndpoint'."

  on main.tf line ***, in resource "azurerm_cognitive_account" "cognitive_service":
 ***: resource "azurerm_cognitive_account" "cognitive_service" {
```

https://github.com/terraform-providers/terraform-provider-azurerm/blob/c916aa6db4142add1760a9b39bc668eb8b1dfcde/azurerm/internal/services/cognitive/resource_arm_cognitive_account.go#L156-L166

Caused by #5778.
The 'qnaRuntimeEndpoint' type is `*string`.
If we assign a `""` using `utils.String` into 'qnaRuntimeEndpoint', we will get a `""` in json by marshaling.
We should use `d.GetOk("qname_runtime_endpoint")` and need to check if exists.
https://github.com/Azure/azure-sdk-for-go/blob/b7d185716507a401293b9f6c303f6e9798c4cb1b/services/cognitiveservices/mgmt/2017-04-18/cognitiveservices/models.go#L258-L268

### Steps to Reproduce

1. `terraform apply`

### References

* #5778

### PR DESCRIPTION
**Pull request by johejo**
fix #6343

### CODE REVIEW

File: azurerm/internal/services/cognitive/resource_arm_cognitive_account.go [hunk 1]
```diff
@@ -169,6 +167,11 @@ func resourceArmCognitiveAccountCreate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("the QnAMaker runtime endpoint `qna_runtime_endpoint` is required when kind is set to `QnAMaker`")
 	}
 
+	qnaRuntimeEndpoint, exists := d.GetOk("qna_runtime_endpoint")
+	if exists {
+		props.Properties.APIProperties.QnaRuntimeEndpoint = utils.String(qnaRuntimeEndpoint.(string))
+	}
```
Comment (tombuildsstuff) [inline azurerm/internal/services/cognitive/resource_arm_cognitive_account.go:173]: could we add [an acceptance test](https://github.com/terraform-providers/terraform-provider-azurerm/blob/304c19a6c07659d20082c0296121accff89471c8/azurerm/internal/services/cognitive/tests/resource_arm_cognitive_account_test.go#L17-L38) to confirm this works when the `qna_runtime_endpoint` field is unset?

Comment (johejo) [reply azurerm/internal/services/cognitive/resource_arm_cognitive_account.go:173]: OK, I will try. This is my first contribution. Is there any existing test code similar to this change I can refer to?

**Comment by johejo** [comment]
Hmm, temporarily I reverted the fix after adding the test, but the test seems to pass. I am confused.

**Comment by katbyte** [comment]
@johejo, could it be the service's API changed behaviour and fix the bug? The tests used to pass with the current code

**Comment by johejo** [comment]
@katbyte 
I read the surrounding code and thought this fix was fine.
However, adding tests, it appears that the bug never existed in the first place.

If the bug report is wrong, close this PR.

**Comment by katbyte** [comment]
closing as requested.

**Pull request closed without merging**